frontend/apps/reader/modules/readerzooming.lua

```diff
@@ -161,14 +199,121 @@ function ReaderZooming:onZoom(direction)
+    })[config.zoom_direction]
+    local zoom_range_number = config.zoom_range_number
+    local zoom_factor = config.zoom_factor
+    local zoom_mode_genus = ({
+        [4] = "page",
+        [3] = "content",
+        [2] = "columns",
+        [1] = "rows",
+        [0] = "manual",
+    })[config.zoom_mode_genus]
+    local zoom_mode_species = ({
+        [2] = "",
+        [1] = "width",
+        [0] = "height",
+    })[config.zoom_mode_species]
+    settings.zoom_overlap_h = config.zoom_overlap_h
+    settings.zoom_overlap_v = config.zoom_overlap_v
+    if btn == "set_zoom_overlap_h" then
+        self:_zoomPanChange(_("Set horizontal overlap"), "zoom_overlap_h")
+        settings.zoom_overlap_h = self.zoom_overlap_h
+    elseif btn == "set_zoom_overlap_v" then
+        self:_zoomPanChange(_("Set vertical overlap"), "zoom_overlap_v")
+        settings.zoom_overlap_v = self.zoom_overlap_v
+    end
+
+    local zoom_mode
+    if zoom_mode_genus == "page" or zoom_mode_genus == "content" then
+        zoom_mode = zoom_mode_genus..zoom_mode_species
+    else
+        zoom_mode = zoom_mode_genus
+        self.ui:handleEvent(Event:new("SetScrollMode", false))
+    end
+    zoom_mode = util.arrayContains(self.available_zoom_modes, zoom_mode) and zoom_mode or self.DEFAULT_ZOOM_MODE
+    settings.zoom_mode = zoom_mode
+
+    if settings.right_to_left then
+        if settings.zoom_bottom_to_top then
+            config.writing_direction = 2
+        else
+            config.writing_direction = 1
+        end
+    else
+        config.writing_direction = 0
+    end
+    settings.right_to_left = nil
+
+    if zoom_mode == "columns" or zoom_mode == "rows" then
+        if btn ~= "columns" and btn ~= "rows" then
+            self.ui:handleEvent(Event:new("SetZoomPan", settings, true))
+            settings.zoom_factor = self:setNumberOf(
+                zoom_mode,
+                zoom_range_number,
+                zoom_mode == "columns" and settings.zoom_overlap_h or settings.zoom_overlap_v
+            )
+        end
+    elseif zoom_mode == "manual" then
+        if btn == "manual" then
+            config.zoom_factor = self:getNumberOf("columns")
+        else
+            self:setNumberOf("columns", zoom_factor)
+        end
+        self.ui:handleEvent(Event:new("SetZoomPan", settings, true))
+    end
+    self.ui:handleEvent(Event:new("SetZoomMode", zoom_mode))
+    if btn == "columns" or btn == "rows" then
+        config.zoom_range_number = self:getNumberOf(
+            zoom_mode,
+            btn == "columns" and settings.zoom_overlap_h or settings.zoom_overlap_v
+        )
+    end
+    if tonumber(btn) then
+        UIManager:show(InfoMessage:new{
+            timeout = 2,
+            text = T(_([[
+    Zoom set to:
+
+    mode: %1
+    number of columns: %2
+    number of rows: %4
+    horizontal overlap: %3 %
+    vertical overlap: %5 %
+    zoom factor: %6.]]),
```

Comment:
Do we need the final period/dot here ? (It feels a bit noisy, `Zoom factor: 1.60.` )

Reply:
Only if it's a sentence. The bigger question is if the presentation is consistent with elsewhere, e.g., I seem to recall it might look closer to

```
Something something:

  First letter capitalized: %1
  Also note the leading spaces: %2
```

NB Only if that's how it's done elsewhere, just whatever's consistent.